res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+    midi.sendShortMsg(0x91, 0x1D, 0x0);
+    midi.sendShortMsg(0x91, 0x2D, 0x0);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1C, 0x0);
+    midi.sendShortMsg(0x92, 0x1D, 0x0);
+    midi.sendShortMsg(0x92, 0x2D, 0x0);
+  }
+  else {
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1E, 0x0);
+    midi.sendShortMsg(0x91, 0x1F, 0x0);
+    midi.sendShortMsg(0x91, 0x2C, 0x7F);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1E, 0x0);
+    midi.sendShortMsg(0x92, 0x1F, 0x0);
+    midi.sendShortMsg(0x92, 0x2C, 0x7F);
+
+    //Do beat alignement only if the tracks are already on Tempo
+    // and only if they are playing
+    if ( (engine.getValue(deckA, "play_latched")) && (engine.getValue(deckB, "play_latched")) ){
+
+      var beat1 = engine.getValue(deckA, "beat_distance");
+      var beat2 = engine.getValue(deckB, "beat_distance");
+      if (doubleTempo){
+        if (tempo1 > tempo2){
+          if (beat2 > 0.5){
+            beat2 -= 0.5;
+          }
+          beat2 *= 2;
+        }
+        else{ //tempo2 >(=) tempo1
+          if (beat1 > 0.5){
+            beat1 -= 0.5;
+          }
+          beat1 *= 2;
+        }
+      }
+      diff = beat1 - beat2;
+      if (diff < 0){
+        diff = 1+diff;
+      }
+      if ((diff < 0.02) || (diff > 1-0.02))
+      {
+        //Deck1
+        midi.sendShortMsg(0x91, 0x1C, 0x0);
+        midi.sendShortMsg(0x91, 0x1D, 0x0);
+        midi.sendShortMsg(0x91, 0x2D, 0x7F);
+        //Deck2
+        midi.sendShortMsg(0x92, 0x1C, 0x0);
+        midi.sendShortMsg(0x92, 0x1D, 0x0);
+        midi.sendShortMsg(0x92, 0x2D, 0x7F);
+      }
+      else if ( diff < 0.5)
+      {
+        //Deck1
+        midi.sendShortMsg(0x91, 0x1C, 0x0);
+        midi.sendShortMsg(0x91, 0x1D, 0x7F);
+        midi.sendShortMsg(0x91, 0x2D, 0x0);
+        //Deck2
+        midi.sendShortMsg(0x92, 0x1D, 0x0);
+        midi.sendShortMsg(0x92, 0x1C, 0x7F);
+        midi.sendShortMsg(0x91, 0x2D, 0x0);
+      }
+      else {
+        //Deck1
+        midi.sendShortMsg(0x91, 0x1D, 0x0);
+        midi.sendShortMsg(0x91, 0x1C, 0x7F);
+        midi.sendShortMsg(0x91, 0x2D, 0x0);
+        //Deck2
+        midi.sendShortMsg(0x92, 0x1C, 0x0);
+        midi.sendShortMsg(0x92, 0x1D, 0x7F);
+        midi.sendShortMsg(0x92, 0x2D, 0x0);
+      }
+    }//if playing
+    else {
+      //Deck1
+      midi.sendShortMsg(0x91, 0x1C, 0x0);
+      midi.sendShortMsg(0x91, 0x1D, 0x0);
+      midi.sendShortMsg(0x91, 0x2D, 0x0);
+      //Deck2
+      midi.sendShortMsg(0x92, 0x1C, 0x0);
+      midi.sendShortMsg(0x92, 0x1D, 0x0);
+      midi.sendShortMsg(0x92, 0x2D, 0x0);
+    }
+  }//else tempo
+};
+
+// After a channel change, make sure we read the current status
+DJCi500.updateDeckStatus = function(group) {
+  var playing = engine.getValue(group, "play_indicator");
+  var volume = script.absoluteLinInverse(engine.getValue(group, "vu_meter"), 0.0, 1.0, 0, 127);
+
+  // Update the vinyl button
+  var vinylState = false;
+  var deckIndex = parseInt(group.charAt(8)) - 1;

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
  var deckIndex = script.deckFromGroup(deckData.currentDeck) - 1;
```